envconfig_derive/src/lib.rs

```diff
@@ -54,16 +54,33 @@ fn impl_envconfig_for_struct(
 
 fn gen_field_assign(field: &Field) -> proc_macro2::TokenStream {
     let attr = fetch_envconfig_attr_from_field(field);
-    let list = fetch_list_from_attr(field, attr);
-    let from_value = find_item_in_list_or_panic(field, &list, "from");
-    let opt_default = find_item_in_list(field, &list, "default");
 
-    let field_type = &field.ty;
+    if let Some(attr) = attr {
+        let list = fetch_list_from_attr(field, attr);
+        let from_value = find_item_in_list_or_panic(field, &list, "from");
+        let opt_default = find_item_in_list(field, &list, "default");
 
-    if to_s(field_type).starts_with("Option ") {
-        gen_field_assign_for_optional_type(field, from_value, opt_default)
+        let field_type = &field.ty;
+
+        if to_s(field_type).starts_with("Option ") {
+            gen_field_assign_for_optional_type(field, from_value, opt_default)
+        } else {
+            gen_field_assign_for_non_optional_type(field, from_value, opt_default)
+        }
     } else {
-        gen_field_assign_for_non_optional_type(field, from_value, opt_default)
+        gen_field_assign_for_struct_type(field)
+    }
+}
+
+fn gen_field_assign_for_struct_type(field: &Field) -> proc_macro2::TokenStream {
+    let ident = &field.ident;
+    match &field.ty {
+        syn::Type::Path(path) => {
+            quote! {
+                #ident: #path :: init()?
+            }
+        }
+        _ => panic!("AAA",),
```

Comment:
Could you please panic with a more meaningful message?

Reply:
Ops. Sorry.
Which is the best message for you here?

Reply:
If I understand this correct, then something like this would be enough:

```rust
panic!(format!("Expected field type to be a path: {:?}", field))
```

The point is to give some clue to developers what they're doing wrong, if they can this panic.

Reply:
addressed